### PR TITLE
Upgrade configs to anemoi-graphs 0.4.1

### PR DESCRIPTION
**Pull request by JPXKQX**
### Pull Request Description

This PR updates the configuration files to align with the new recommended `anemoi-graphs` version (0.4.1). While the old configuration files are still supported in this release, they will be deprecated in a future version.  

**Action Required**: Please update your local configuration files to avoid future compatibility issues.

Tested on:
- 1 GPU: 
   - [x] LAM (datashader=False) with old & new graph config files.
   - [x] Stretched (datashader=False) with old & new graph config files.
   - [ ] LAM (datashader=True). This will be solved with PR #152 .
   - [ ] Stretched(datashader=True). This will be solved with PR #152 .

---

### Key Changes

1. **Field Renaming**:  
   - `edge_builder` has been renamed to `edge_builders`.  
   - The new `edge_builders` field now accepts a **list of edge builders**.

2. **Masking Specification**:  
   - The `source_mask_attr_name` and `target_mask_attr_name` fields are now defined **under each individual edge builder**.

### CODE REVIEW

**Review by HCookie — DISMISSED** [review]

**Review by HCookie — APPROVED** [review]

File: src/anemoi/training/diagnostics/callbacks/plot.py [hunk 1]
```diff
@@ -938,7 +938,7 @@ def _plot(
             torch.cat(tuple(x[self.sample_idx : self.sample_idx + 1, ...].cpu() for x in outputs[1])),
             in_place=False,
         )
-        output_tensor = pl_module.output_mask.apply(output_tensor, dim=2, fill_value=np.nan).numpy()
+        output_tensor = pl_module.output_mask.apply(output_tensor, dim=1, fill_value=np.nan).numpy()
```
Comment (mchantry) [inline src/anemoi/training/diagnostics/callbacks/plot.py:941]: Why does this dim change?

Comment (JPXKQX) [reply src/anemoi/training/diagnostics/callbacks/plot.py:941]: It didn't work (LAM + `plot: detailed`). It seems that the `output_tensor` changed its shape at some point.
